src/routes/preHandlers/courses.js

```diff
@@ -53,7 +54,15 @@ exports.authorizeGetCourseList = async (req) => {
 
   const courseTrainerId = get(req, 'query.trainer');
   const courseCompanyId = get(req, 'query.company');
-  this.checkAuthorization(courseTrainerId, courseCompanyId, credentials);
+  const traineeId = get(req, 'query.trainees');
+
+  let traineeCompanyId = null;
+  if (traineeId) {
+    const trainee = await User.findOne({ _id: traineeId }).lean();
+    traineeCompanyId = trainee.company._id.toHexString();
+  }
+
+  this.checkAuthorization(credentials, courseTrainerId, courseCompanyId, traineeCompanyId);
```

Comment:
tu peux faire this.checkAuthorization(credentials, courseTrainerId, courseCompanyId, **traineeCompanyId || null**). ça permettra de passer null en argument si traineeCompanyId est undefined et ça t'évite de déclarer traineeCompanyId sans l'utiliser après.

Reply:
Je crois que si une variable n'est pas defini avant de l'appeler ca renvoi une erreur, jsp si avec une autre version de js ca se passe autrement

Reply:
C'est vrai pardon !